node/forest_libp2p/src/service.rs

```diff
@@ -716,44 +737,105 @@ async fn handle_forest_behaviour_event<DB, P: StoreParams>(
-            RequestResponseEvent::OutboundFailure {
-                peer,
-                request_id,
-                error,
-            } => {
-                warn!(
-                    "ChainExchange outbound error (peer: {:?}) (id: {:?}): {:?}",
-                    peer, request_id, error
-                );
+            }
+        }
+        RequestResponseEvent::OutboundFailure {
+            peer,
+            request_id,
+            error,
+        } => {
+            warn!(
+                "ChainExchange outbound error (peer: {:?}) (id: {:?}): {:?}",
+                peer, request_id, error
+            );
 
-                let tx = cx_request_table.remove(&request_id);
+            let tx = cx_request_table.remove(&request_id);
 
-                // Send error through channel out.
-                if let Some(tx) = tx {
-                    if tx.send(Err(error.into())).is_err() {
-                        warn!("RPCResponse receive failed")
-                    }
+            // Send error through channel out.
+            if let Some(tx) = tx {
+                if tx.send(Err(error.into())).is_err() {
+                    warn!("RPCResponse receive failed")
                 }
             }
-            RequestResponseEvent::InboundFailure {
-                peer,
-                error,
-                request_id: _,
-            } => {
-                debug!(
-                    "ChainExchange inbound error (peer: {:?}): {:?}",
-                    peer, error
-                );
-            }
-            RequestResponseEvent::ResponseSent { request_id, .. } => {
-                emit_event(
-                    network_sender_out,
-                    NetworkEvent::ChainExchangeResponseOutbound { request_id },
-                )
-                .await;
-            }
-        },
+        }
+        RequestResponseEvent::InboundFailure {
+            peer,
+            error,
+            request_id: _,
+        } => {
+            debug!(
+                "ChainExchange inbound error (peer: {:?}): {:?}",
+                peer, error
+            );
+        }
+        RequestResponseEvent::ResponseSent { request_id, .. } => {
+            emit_event(
+                network_sender_out,
+                NetworkEvent::ChainExchangeResponseOutbound { request_id },
+            )
+            .await;
+        }
+    }
+}
+
+#[allow(clippy::too_many_arguments)]
+async fn handle_forest_behaviour_event<DB, P: StoreParams>(
+    swarm: &mut Swarm<ForestBehaviour<P>>,
+    event: ForestBehaviourEvent<P>,
+    db: &Arc<ChainStore<DB>>,
+    genesis_cid: &Cid,
+    network_sender_out: &flume::Sender<NetworkEvent>,
+    hello_request_table: &mut HashMap<
+        RequestId,
+        futures::channel::oneshot::Sender<Result<HelloResponse, RequestResponseError>>,
+    >,
+    cx_request_table: &mut HashMap<
+        RequestId,
+        futures::channel::oneshot::Sender<Result<ChainExchangeResponse, RequestResponseError>>,
+    >,
```

Comment:
The type signatures for `hello_request_table` and `cx_request_table` are repeated three times. Would probably be worth it to use a type synonym.

Reply:
Done